UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs

```diff
@@ -124,13 +126,70 @@ public async Task<SearchBatchResponse> GetBatchInfoBasedOnProducts(List<Products
                 }
                 else
                 {
-                    logger.LogInformation(EventIds.QueryFileShareServiceNonOkResponse.ToEventId(), "File share service with uri {RequestUri}, responded with {StatusCode} and _X-Correlation-ID:{correlationId}", httpResponse.RequestMessage.RequestUri, httpResponse.StatusCode, correlationId);
+                    logger.LogError(EventIds.QueryFileShareServiceENCFilesNonOkResponse.ToEventId(), "Error in file share service for query search ENC files with uri:{RequestUri}, responded with {StatusCode} and BatchId:{batchId} and _X-Correlation-ID:{correlationId}", httpResponse.RequestMessage.RequestUri, httpResponse.StatusCode, batchId, correlationId);
                 }
-            } while (httpResponse.IsSuccessStatusCode && internalSearchBatchResponse.Entries.Count != 0 && internalSearchBatchResponse.Entries.Count < prodCount);
-
+            } while (httpResponse.IsSuccessStatusCode && internalSearchBatchResponse.Entries.Count != 0 && internalSearchBatchResponse.Entries.Count < prodCount && !string.IsNullOrWhiteSpace(uri));
+            CheckProductsExistsInFss(products, correlationId, internalSearchBatchResponse, internalNotFoundProducts, prodCount);
             return internalSearchBatchResponse;
         }
 
+        private void CheckProductsExistsInFss(List<Products> products, string correlationId, SearchBatchResponse internalSearchBatchResponse, List<Products> internalNotFoundProducts, int prodCount)
+        {
+            if (internalSearchBatchResponse.Entries.Any() && prodCount != internalSearchBatchResponse.Entries.Count)
+            {
+                List<Products> internalProducts = new List<Products>();
+                ConvertFssSearchBatchResponseToProductResponse(internalSearchBatchResponse, internalProducts);
+                GetProductDetailsNotFoundInFss(products, internalNotFoundProducts, internalProducts);
+            }
+            if (internalNotFoundProducts.Any() || !internalSearchBatchResponse.Entries.Any())
+            {
+                var internalNotFoundProductsPayLoadJson = JsonConvert.SerializeObject(internalNotFoundProducts.Any() ? internalNotFoundProducts.Distinct() : products);
+                logger.LogError(EventIds.FSSResponseNotFoundForRespectiveProductWhileQuering.ToEventId(), "File share service response not found while quering to FSS {internalNotFoundProductsPayLoadJson} and _X-Correlation-ID:{correlationId}", internalNotFoundProductsPayLoadJson, correlationId);
+                throw new FulfilmentException(EventIds.FSSResponseNotFoundForRespectiveProductWhileQuering.ToEventId());
+            }
+        }
+
+        private void GetProductDetailsNotFoundInFss(List<Products> products, List<Products> internalNotFoundProducts, List<Products> internalProducts)
```

Comment:
Kindly change method name from CheckProductsExistsInFss to CheckProductsExistsInFileShareService and GetProductDetailsNotFoundInFss to GetProductDetailsNotFoundInFileShareService

Also change message in logerror at line 147 to Error in file share service while searching ENC files and no data found while querying for products. Log batch id as well

Reply:
Resolved